portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java

```diff
@@ -27,18 +27,23 @@
 import javax.servlet.http.HttpServletResponse;
 
 /** Servlet responsible for deleting comments. */
-@WebServlet("/delete-data")
+@WebServlet("/delete-comment")
 public class DeleteCommentServlet extends HttpServlet {
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("Comment");
+    long id = Long.parseLong(request.getParameter("id"));
+
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("Comment").setKeysOnly();
```

Comment:
Since you the id (which is the key) you should delete the entity directly. See for example https://stackoverflow.com/q/22702822

you would do something like 

Key key = KeyFactory.createKey("Comment", id);
datastore.delete(key);

You should put this inside a try catch block and handle any exception.

Reply:
Thanks, made these changes